index.js

```diff
@@ -9,13 +9,27 @@ exports.prepare = function (extensions, filepath, cwd, nothrow) {
   var attempts = [];
   var err;
   var onlyErrors = false;
-  var ext = extension(filepath);
-  if (Object.keys(require.extensions).indexOf(ext) !== -1) {
+  var exts = extension(filepath);
+
+  var supportsExtension = Object.keys(require.extensions).indexOf(exts[0]) !== -1;
```

Comment:
actually, i don't think require.extensions will ever have something like `.babel.js` because node's loader doesn't work with that.  hmmm, this is a tricky check